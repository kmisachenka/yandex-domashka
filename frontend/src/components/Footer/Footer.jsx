import React from 'react';
import cn from 'classnames';
import * as PropTypes from 'prop-types';

import styles from './Footer.module.scss';

export default function Footer(props) {
  const { copyright, className } = props;
  return (
    <footer className={cn(className, styles.footer)}>
      <div className={styles['show-on-desktop']} />
      <div className={styles.name}>{copyright}</div>
      <div>© Яндекс ШРИ</div>
    </footer>
  );
}

Footer.defaultProps = {
  copyright: 'Вася Пупкин',
  className: '',
};

Footer.propTypes = {
  copyright: PropTypes.string,
  className: PropTypes.string,
};
