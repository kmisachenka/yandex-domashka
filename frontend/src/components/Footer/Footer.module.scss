@import '../../media';

.footer {
  align-items: center;

  background: #fff;

  color: rgba(51, 51, 51, 0.6);
  display: flex;

  font-family: Helvetica, serif;
  font-size: 16px;
  justify-content: space-between;
  line-height: 32px;

  min-height: 54px;

  padding-left: 16px;
  padding-right: 16px;
  text-align: right;

  @include desktop {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
  }
}

.name {
  @include desktop {
    justify-self: center;
  }
}

.show-on-desktop {
  display: none;

  @include desktop {
    display: block;
  }
}
