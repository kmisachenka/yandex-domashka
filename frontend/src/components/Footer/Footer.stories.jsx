import React from 'react';
import { storiesOf } from '@storybook/react';
import Footer from './Footer';

storiesOf('Блок: Футер', module)
  .add('футер', () => (
    <Footer />
  ));
