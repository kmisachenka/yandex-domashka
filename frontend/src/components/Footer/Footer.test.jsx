import React from 'react';
import { shallow } from 'enzyme';

import Footer from './Footer';

describe('Footer', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render with custom copyright', () => {
    const wrapper = shallow(<Footer copyright="Кирилл Мисоченко" />);
    const name = wrapper.find('.name');
    expect(name).toHaveText('Кирилл Мисоченко');
  });
  it('should apply custom classes', () => {
    const wrapper = shallow(<Footer className="custom" />);
    expect(wrapper).toHaveClassName('custom');
  });
});
