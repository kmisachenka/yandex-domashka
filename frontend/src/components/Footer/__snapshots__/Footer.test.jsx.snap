// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
<footer
  className="footer"
>
  <div
    className="show-on-desktop"
  />
  <div
    className="name"
  >
    Вася Пупкин
  </div>
  <div>
    © Яндекс ШРИ
  </div>
</footer>
`;
